src/include/messenger/messenger.h

```diff
@@ -356,10 +365,16 @@ class Messenger : public common::DedicatedThreadTask {
   std::atomic<message_id_t> next_message_id_{0};
   /** The source callback ID that gets automatically prefixed to messages. */
   std::atomic<callback_id_t> next_callback_id_{static_cast<uint8_t>(BuiltinCallback::NUM_BUILTIN_CALLBACKS) + 1};
-  /** The ID of the next listening router to be made from ListenForConnection(). */
+  /**
+   * The ID of the next listening router to be made from ListenForConnection().
+   * Note that this is always accessed while holding routers_add_mutex_.
+   */
   std::atomic<router_id_t> next_router_id_{0};
```

Comment:
Is there a reason that this is atomic but `next_connection_id_` isn't? (Unless I'm misunderstanding I think you may have just forgotten to update this).

Reply:
yep, you're absolutely right